### PR TITLE
Bug fix for #417

### PR DESCRIPTION
**Pull request by matthias-ronge**
Bug fix for #417

### CODE REVIEW

**Comment by stweil** [comment]
Please squash the first two commits.

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java [hunk 1]
```diff
@@ -614,17 +614,81 @@ private static DocStruct getOrCreateChild(DocStruct parent, String type, String
 		try {
 			return parent.getChild(type, identifierField, identifier);
 		} catch (NoSuchElementException nose) {
-			DocStruct child = parent.createChild(type, act, ruleset);
+			DocStruct child = act.createDocStruct(ruleset.getDocStrctTypeByName(type));
 			child.addMetadata(identifierField, identifier);
 			try {
 				child.addMetadata(optionalField, identifier);
 			} catch (MetadataTypeNotAllowedException e) {
+				logger.warn(e.getMessage());
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: Other log statements for warning from you are surrounded by `logger.isEnabled...` calls. Did you miss this line?

Comment (matthias-ronge) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: `logger.isEnabled` does only make sense if the logging code does something, like `String` composition. Here the work has already been done so there is nothing worth skipping.

**Comment by matthias-ronge** [comment]
> Please squash the first two commits.

What is the use of it?

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java (hunk 1 above)
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: Ok.

**Comment by stweil** [comment]
@matthias-ronge, here are some links which help to understand the concept:

https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit
http://jamescooke.info/git-to-squash-or-not-to-squash.html
http://programmers.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests
https://www.google.de/search?q=squash+git+commits

Im short, squashing means a little bit more work for you and a little bit less work for everybody else.

**Comment by matthias-ronge** [comment]
@henning-gerhardt In general it would be helpful to name the commits to be squashed by the name (just copy paste), bc. which is ‘first’ depends on the order, in my Git hierarchy, the ”first” and “second” here was the “third-last” and “second-last”, a bit into the middle of of a long history. And, I am not clear if one mis-smashes commits if it is possible to undo that

**Comment by henning-gerhardt** [comment]
Squashing commits was proposed by @stweil and not me. 

If a developer use squashing commits then he is responsible for missing / mis-smashed commits as this is not visible in a pull request.

I will merge this pull today.

**Comment by matthias-ronge** [comment]
Oh really. I got confused by the new icon ;)

**Pull request merged**